src/precice/impl/ParticipantImpl.cpp

```diff
@@ -275,9 +278,11 @@ void ParticipantImpl::initialize()
       PRECICE_DEBUG("Established primary connection {} {}", (requesting ? "from " : "to "), bm2n.remoteName);
     }
   }
+  e2.stop();
 
   PRECICE_INFO("Primary ranks are connected");
 
+  Event e3("repartitioning");
```

Comment:
This event contains a lot, but splitting it is difficult to explain as functionality depends on 2 vs 1 LI.